dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -28,6 +40,232 @@ extern const int BAD_ARGUMENTS;
 extern const int NOT_IMPLEMENTED;
 } // namespace ErrorCodes
 
+namespace
+{
+template <typename T>
+consteval bool checkIfSimpleNumericType()
+{
+    return std::is_integral_v<T> || std::is_floating_point_v<T>;
+}
+
+template <typename T>
+consteval bool checkIfDecimalFieldType()
+{
+    return std::is_same_v<
+               T,
+               DecimalField<
+                   Decimal32>> || std::is_same_v<T, DecimalField<Decimal64>> || std::is_same_v<T, DecimalField<Decimal128>> || std::is_same_v<T, DecimalField<Decimal256>>;
+}
+
+template <typename LeftType, typename RightType>
+bool lessEqual(LeftType left, RightType right)
+{
+    if constexpr (checkIfDecimalFieldType<LeftType>() && checkIfDecimalFieldType<RightType>())
+    {
+        // This `if` branch is necessary or the next branch's codes will be generated
+        // when these two types are all DecimalField type which will cause compilation errors.
+        return left <= right;
+    }
+    else if constexpr (checkIfDecimalFieldType<LeftType>() || checkIfDecimalFieldType<RightType>())
+    {
+        // Only one type will be Decimal as Decimal vs Decimal can be handled in the `else` branch
```

Comment:
The comment should be `as Decimal vs Decimal is handled before`?

Reply:
> 

The code hasn't been updated when reviewing, and the comment here has been deleted.